components/camel-salesforce/camel-salesforce-component/src/main/java/org/apache/camel/component/salesforce/internal/client/AbstractClientBase.java

```diff
@@ -87,6 +90,10 @@ public void start() throws Exception {
 
     @Override
     public void stop() throws Exception {
+        if (!inflightRequests.isTerminated()) {
+            inflightRequests.arriveAndAwaitAdvance();
+        }
+
```

Comment:
I don't really know if there exists a reason, as the Jetty contract states that `onComplete` is called no matter what, but there could be many unknown reasons for this so the new version timeouts after 10 seconds. Now we have tests to prove that also.